src/Axe/Managers/CompileManager.cs

```diff
@@ -0,0 +1,437 @@
+                        case SupportedType.Byte:
+                        case SupportedType.Sbyte:
+                        case SupportedType.Short:
+                        case SupportedType.Ushort:
+                        case SupportedType.Int:
+                        case SupportedType.Uint:
+                        case SupportedType.Long:
+                        case SupportedType.Ulong:
+                        case SupportedType.Double:
+                        case SupportedType.Float:
+                        case SupportedType.Decimal:
+                        case SupportedType.Char:
+                        case SupportedType.String:
+                            {
+                                codeBlock.VerificationCode +=
+                                @"if (result_#s#_#INDEX# != #OUTPUT#)
+                                resultsOfTestCases_#s#[#INDEX#] = false;";
+                                break;
+                            }
+                        case SupportedType.BoolArray:
+                        case SupportedType.ByteArray:
+                        case SupportedType.SbyteArray:
+                        case SupportedType.ShortArray:
+                        case SupportedType.UshortArray:
+                        case SupportedType.IntArray:
+                        case SupportedType.UintArray:
+                        case SupportedType.LongArray:
+                        case SupportedType.UlongArray:
+                        case SupportedType.DoubleArray:
+                        case SupportedType.FloatArray:
+                        case SupportedType.DecimalArray:
+                        case SupportedType.CharArray:
+                        case SupportedType.StringArray:
+                            {
+                                codeBlock.VerificationCode +=
+                                @"var array_#s#_#INDEX# = #OUTPUT#;
+                            if (array_#s#_#INDEX#.Length != result_#s#_#INDEX#.Length)
+                                resultsOfTestCases_#s#[#INDEX#] = false;
+                            else
+                                for (int i_#s# = 0; i_#s# < array_#s#_#INDEX#.Length; i_#s#++)
+                                {
+                                    if (array_#s#_#INDEX#[i_#s#] != result_#s#_#INDEX#[i_#s#])
+                                    resultsOfTestCases_#s#[#INDEX#] = false;
+                                }";
+                                break;
+                            }
+                        default:
+                            break;
+                    }
+
+                    codeBlock.VerificationCode = codeBlock.VerificationCode
+                        .Replace("#INDEX#", i.ToString())
+                        .Replace("#OUTPUT#", codeBlock.TestCases[i].Output)
+                        .Replace("#INPUT#", codeBlock.TestCases[i].Input);
+                }
+
+                switch (codeBlock.OutputType)
```

Comment:
a lot of duplicated code in each `case`. there must be a shorter way to map SupportedType values to type name, e.g: read all SupportedType enum values, get `[Desplay]` attribute, store them in `Dictionary<SupportedType, string>` . then retrive a string based on `codeBlock.OutputType` and do `codeBlock.VerificationCode = codeBlock.VerificationCode.Replace(...)`  once

Reply:
I fixed it. Now, I use Display attribute of SupportedType enum. See it in the commit: https://github.com/AleksandrSharykin/Axe/pull/19/commits/807c0047976b0e9035738b1a37e7094f6dc8b4b8